### ISSUE TITLE
[Accessibility]: Changeset sync button is not an actual button

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#35007]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34089

### Problem description

The sync icon next to "Last synced X minutes ago." is not addressable using keyboard navigation, as it's not a proper button.

Also, it's too small for mobile probably. 

### Expected behavior

It can be focussed using keyboard navigation and clicked on accordingly. 

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@eseliger Any objections to assigning this to GitStart to work on?

**Comment by eseliger** [comment]
nope, go for it!

**Comment by umpox** [comment]
> Also, it's too small for mobile probably.

@gitstart-sourcegraph For this issue let's focus primarily on making this icon focusable. I think there's a wider scope of work with how we use icons with tooltips and if that is suitability for mobile. Discussing with design about future plans for usage of tooltips overall
